### ISSUE TITLE
Volume mapping making directories out of files

### ISSUE THREAD

**Issue by stilren** [docker/for-win#4746]
### Expected behavior
Expected behavior
A volume mapping in my docker-compose.yml results in the exact structure of the directory I map, being available at the specified location on the container.

### Actual behavior
When mapping like this:

`volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf`

a folder in the host called ngnix.conf is created during docker-compose up and there is a crash with the error message:
ERROR: for 5567d9cbeb1b_production_nginx  Cannot start service nginx: OCI runtime create failed: container_linux.go:345: starting container process caused "process_linux.go:430: container init caused \"rootfs_linux.go:58: mounting \\\"/host_mnt/c/path/to/config/nginx.conf\\\" to rootfs \\\"/var/lib/docker/overlay2/0bd2c0d36cd19750b599858bd005941a074aebfb9068a307828fd740d175d922/merged\\\" at \\\"/var/lib/docker/overlay2/0bd2c0d36cd19750b599858bd005941a074aebfb9068a307828fd740d175d922/merged/etc/nginx/nginx.conf\\\" caused \\\"not a directory\\\"\"": unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

### Information
Please see https://github.com/docker/for-win/issues/1501 for a closed version of this issue. (I have tried the solutions put forward in that thread)

  - Windows Version: Version	10.0.17134 Build 17134 (win 10 pro)
  - Docker Desktop Version: 2.1.0.3 (38240)
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

**Comment by judos** [comment]
I got the same issue
Docker version 19.03.4, build 9013bf583a

```
/ # ls -lah
drwxr-xr-x    2 root     root          40 Oct 31 23:15 application.properties
```

docker-compose.yml
```
version: '3.3'
services:
  immo:
    build:
      context: .
      dockerfile: Dockerfile
    volumes:
      - /home/environments/app/application.properties:/application.properties
      - /home/environments/app/backup:/backup-files
    expose:
      - 3307
    ports:
      - "${port}:8080"
    restart: unless-stopped
```

Long syntax also doesnt work:
```
    volumes:
      - type: bind
        source: /home/environments/app/application.properties
        target: /application.properties
```

**Comment by ian-wandisco** [comment]
+1 - Any thoughts on working round this much appreciated.

**Comment by judos** [comment]
At the moment I just copy the file into the image (in the Dockerfile "COPY ./application.properties /application.properties")

Everything I tried with the mapping itself didn't work.

**Comment by aznarepse** [comment]
This happens also in Linux (CentOS) if the host path does not exist at the time of running the container.
It seems that the default for docker is to create a directory if the path does not exists (https://github.com/docker/for-win/issues/1501). The only work around for me is to make sure that the host path and file exist before the run.

**Comment by judos** [comment]
On Ubuntu I had no more issues so far, so I switching OS worked out for me.

**Comment by arunvc** [comment]
Same issue in ubuntu 18, 
if there is no file in the host volume, docker create a folder with the file name, docker fails to recognize a file and folder.

work around is
1. delete the folder in host volume with your file name.
2. copy the file to host volume and restart the container.

Or before starting the container place your file at the host volume

**Comment by kjeldflarup** [comment]
I just upgraded my PC installed docker

This used to work
```
      volumes:
          - ~/.ssh/id_scantron:/id_scantron
```
Now it does not.  
Has anything changed in docker defaults
```
$ sudo docker --version
Docker version 19.03.8, build afacb8b7f0
```

On another server it works
```
$ docker --version
Docker version 18.06.0-ce, build 0ffa825
```
So what has changed between version 18 and 19?

**Comment by kjeldflarup** [comment]
Maybe I found an explanation to my problem. I used to run docker with sudo. But after running this command
```
 sudo usermod -aG docker your-user
```
Then it worked. The reason seems to be that `~/.ssh/id_scantron` is now mapped to `/root/.ssh/id_scantron` which does not exist, and thus docker creates  `/root/.ssh/id_scantron` as a folder and maps it.

Thus the change seem to be related to docker-compose (or sudo).
